ls1intum/Artemis
# `Development`: Ensure the loading of quiz exercises for a student exam within the integration tests

### Checklist
#### General
#### Server

### Motivation and Context
PR #5458 hot-fixed an issue, where quiz exercises were not loaded for an student exam. Loading the quiz exercises involves a two-step process. First, the quiz exercise is loaded via the `StudentExamResource.getWithExercisesById()`, but this is only the "container" that contains quiz questions. In the second step, the actual quiz-questions must be loaded from the `quizExerciseRepository`.

### Description
Within the `StudentExamIntegrationTest` in suitable places, where the quiz tasks are already tested in detail, a check for the presence of the quiz tasks was added. This ensures, that not only the "container", but also the actual quiz questions are present. 

### Steps for Testing
- Take a look at the code changes and verify, that the Tests are passing
- In your local IDE, you could comment out the `loadQuizExercisesForStudentExam` within the `StudentExamResource`. Verify, that the tests fail.


### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| StudentExamResource.java | 76 % | 93 % |
